### DISCUSSION TITLE
⛵ Charting Our Course - Actions Community Check-in

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#58226] [Actions]
If you're moving to GitHub.com and excited about all of the great features developers love—from GitHub Actions to GitHub Codespaces—you’ll have existing data that you want to bring with you. This week we launched another great feature, [GitHub Enterprise Importer](https://github.blog/changelog/2023-06-12-introducing-github-enterprise-importer), a self-serve tool which empowers teams to migrate their code, history, and collaboration context to GitHub Enterprise Cloud so your team doesn’t miss a beat and can get right to enjoying those great features.

Read more about it on this [detailed blog post](https://github.blog/2023-06-12-cloud-migration-made-easy-introducing-github-enterprise-importer/) and if you have any questions or feedback, post them to [this discussion](https://github.com/orgs/community/discussions/55519). 

That’s not it, we also shipped some great Actions updates to help you automate faster. Before we dive into them, let’s get started by ✨sparkling some of our amazing community members who stood out in the Actions space. 

## ✨ Meet the Community Champions: Hear the Thunderous Shoutouts! 📢

- A BIG round of applause 👏 for our superstar [@jge162](https://github.com/jge162) being a top contributor 💗  in the Actions space. Take a look at some of their top responses:
  - Shared the missing workflow_dispatch code with [@ZedZipDev](https://github.com/ZedZipDev) and that helped them [run the workflow](https://github.com/orgs/community/discussions/58022#discussioncomment-6174243).
  - Thank You for sharing the syntax with [@vansh2552](https://github.com/vansh2552) and helping them [Invoke a workflow inside another workflow](https://github.com/orgs/community/discussions/58000)
- Awesome job ✨ by [@DSDmark](https://github.com/DSDmark) for sharing a [descriptive answer](https://github.com/orgs/community/discussions/57945) and unblocking [@ZedZipDev](https://github.com/ZedZipDev) to create .github/workflows in their Github Repository.
- [Saved the day](https://github.com/orgs/community/discussions/57923) for his fellow community member [@paul-sheridan](https://github.com/paul-sheridan) - Amazing work by [@dumitrux](https://github.com/dumitrux) showcasing their expertise and [resolving a tough issue](https://github.com/orgs/community/discussions/57923). 💯

✅ Mark as Answer: Don’t forget to click the ‘Mark as Answer’ button when someone solves your question to help others who may have the same question ✅

### [GitHub Actions: All Actions will run on Node16 instead of Node12 by default](https://github.blog/changelog/2023-06-13-github-actions-all-actions-will-run-on-node16-instead-of-node12-by-default/) 👀 

We have started the deprecation process of Node12 for GitHub Actions. We plan to migrate all actions to run on Node16 by Summer 2023. 

- For Actions maintainers: Update your actions to run on Node16 instead of Node12 (Actions configuration settings)
- For Actions users: Update your workflows with latest versions of the actions which runs on Node16 (Using versions for Actions)

To opt out of this change and continue using Node12, please [read the changelog](https://github.blog/changelog/2023-06-13-github-actions-all-actions-will-run-on-node16-instead-of-node12-by-default/). 

### [You can now disable repo level self-hosted runners in an Enterprise and Organization](https://github.blog/changelog/2023-06-13-github-actions-you-can-now-disable-repo-level-self-hosted-runners-in-an-enterprise-and-organization) 🆕 

Enterprise administrators - You can now disable repository level self-hosted runners across organizations and enterprise user namespaces.

Once this new setting has been enabled users will no longer be able to register new self-hosted runners in a repository and existing runners will not be able to receive new jobs.

Learn more about [Disabling or limiting GitHub Actions for your organization](https://docs.github.com/en/organizations/managing-organization-settings/disabling-or-limiting-github-actions-for-your-organization)

### [GitHub Actions – Just-in-time self-hosted runners](https://github.blog/changelog/2023-06-02-github-actions-just-in-time-self-hosted-runners) 🏃 

You can now create single-use self-hosted runners without time-limited registration tokens using the REST API.

[Learn more about just-in-time runners](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions)

## ☝️ GitHub Availability Report ➕

At GitHub, we put developers first and work hard to provide a safe, open, and inclusive platform for code collaboration. We are committed to be as transparent as we can and here to provide you updates on any disruption of services and with that in mind, we publish GitHub availability reports each month owning and explaining what happened and how we mitigated each specific incident. 

We update our [status page](https://www.githubstatus.com/) in real time in case of an ongoing incident about Actions. If you see an open incident, this means that we are looking into it already. If not, please let us know about it in the [GitHub community](https://github.com/orgs/community/discussions/categories/actions).

## 🤝  Help us build your Actions know-how [Join the FAQ](https://github.com/orgs/community/discussions/48283)❓

Are you constantly grappling with questions and occasional challenges while working with Actions? Share your queries and common issues [in this discussion](https://github.com/orgs/community/discussions/48283), and let's collaboratively build an invaluable resource packed with top Actions FAQs. 

Together, we can create an evergreen reference that addresses your concerns and empowers the Actions community. Let's join our hands in shaping a knowledge hub that ensures smooth sailing in your Actions 🚢 endeavors!

That's all for now folks. Got feedback or feature requests for anything Actions? ❓Post them on [community](https://github.com/orgs/community/discussions/categories/actions) and share enhancements you’d like to see on the product. Till next time! 🚀

**Comment by MikeMcC399** [comment]
@Akash1134 
Can you comment on the future of GitHub JavaScript Actions after Node.js 16 enters [end-of-life](https://github.com/nodejs/release#release-schedule) in September 2023? (Also mentioned in https://github.com/orgs/community/discussions/53217.)

**Reply by Akash1134** [reply]
Hi @MikeMcC399 , I will check that with our team internally and update. Thanks!

**Reply by MikeMcC399** [reply]
Thank you @Akash1134!
We urgently need JavaScript Actions to support at least Node.js `18`, the LTS version, so API calls which need to run under Node.js `18` have the environment they need. This is independent of the upcoming end-of-life of Node.js `16`.

At the moment workarounds are being used. Ultimately if GitHub JavaScript Actions support for later Node.js versions is unable to be provided, then other solutions would have to be looked for, so I hope that we don't have to deal with this situation and that GitHub JavaScript Actions will provide the needed support of `node18`,  `node20` or both in a timely manner.

Please provide an update.

**Reply by MikeMcC399** [reply]
- This is related to a enhancement request for Cypress. See https://github.com/cypress-io/github-action/issues/968

**Reply by Akash1134** [reply]
Hi @MikeMcC399 , Sorry about the delay in my response. We have an update for you. Please refer [actions/runner nodejs plan and next steps](https://github.com/actions/runner/discussions/2704) If you have further queries, feel free to ask our team in the discussion. Thanks!

**Reply by MikeMcC399** [reply]
Hi @Akash1134 Thank you for the commitment to supporting `node20`. We will look forward to the availability!

**Reply by MikeMcC399** [reply]
See https://github.com/actions/runner/issues/2619#issuecomment-1679003443

> Hello, latest runner release ([2.308.0](https://github.com/actions/runner/releases/tag/v2.308.0)) contains latest node16 version and node20 version. This runner version will be fully rolled-out by the end of this week. After release you will be able to use `'node20'` actions

**Comment by crltsmrtnz** [comment]
🙌
